src/translator/response.h

```diff
@@ -65,10 +53,10 @@ struct Response {
   /// source or target.
   std::vector<SentenceQualityScore> qualityScores;
 
-  /// Alignments between source and target. Each Alignment is a
-  /// sparse matrix representation with indices corresponding
-  /// to (sub-)words accessible through Annotation.
-  std::vector<Alignment> alignments;
+  /// Alignments between source and target. This is a collection of dense matrices providing
+  ///    P[t][s] = p(source-token s  | target token t)
+  /// with an alignment matrix for each sentence.
+  std::vector<marian::data::SoftAlignment> alignments;
```

Comment:
`std::vector<std::vector<float> >` for better or worse

Reply:
You mean `std::vector<std::vector<std::vector<float>>>` (I'm sorry....)? This is `#sentences * # source-tokens * #target-tokens` (3D Tensor)

Reply:
Yes, what I meant is you shouldn't refer to Marian class names, just substitute what it is defined as.